projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -186,18 +167,18 @@ export class ExplorerComponent {
     this.filters = [...newFilters];
   }
 
-  private getOrDefaultContextItemFromQueryParam(value?: ScopeQueryParam): ContextToggleItem {
+  private getOrDefaultContextItemFromQueryParam(value: ValueOrNull<string>): ExplorerContextToggleItem {
     return this.contextItems.find(item => value === item.value.scopeQueryParam) || this.contextItems[0];
   }
 
-  private getQueryParamFromContext(context: ExplorerGeneratedDashboardContext): ScopeQueryParam {
+  protected getQueryParamFromContext(context: string): string {
     switch (context) {
       case ObservabilityTraceType.Api:
         return ScopeQueryParam.EndpointTraces;
       case 'SPAN':
         return ScopeQueryParam.Spans;
       default:
-        return assertUnreachable(context);
+        return this.contextItems[0].value.scopeQueryParam;
```

Comment:
rather than defaulting to first, could we change the whole impl of this method to search through the context items for the matching context and return the associated query param? This is basically what we do in the method right above going in the other direction from query param to context.

Reply:
done